lib/reek/smells/uncommunicative_variable_name.rb

```diff
@@ -61,10 +62,15 @@ def sniff(ctx)
         end
       end
 
-      def bad_name?(name, _ctx)
-        var = name.to_s.gsub(/^[@\*\&]*/, '')
-        return false if accept_names.include?(var)
-        reject_names.find { |patt| patt =~ var }
+      private
+
+      def uncommunicative_variable_name?(name)
+        !acceptable_name?(name)
+      end
+
+      def acceptable_name?(name)
+        Array(accept_names).any? { |accept_pattern| name.match accept_pattern } ||
```

Comment:
This is now **almost** the same as we do in all other Uncommunicative* detectors with the exception of the "Array()" call (which we need if the user gives us literal values). 
I intend to remove the "Array()" call again when I tackle #1066